src/lib/fstab/tsort.rb

```diff
@@ -0,0 +1,37 @@
+require "tsort"
+
+# Namespace for code related to /etc/fstab
+module Fstab
+  # Simple module implementing topological sorting for fstab entries
+  module TSort
+    extend Yast::Logger
+
+    # Sorts a list of fstab entries by the topological order of the mount points
+    def self.sort(list)
+      # Define the two iterators required by TSort
+      each_node = ->(&b) { list.reverse.each(&b) }
```

Comment:
@jreidinger I'm not sure about your proposed solution (doesn't seem to work out of the box). But this works and should address your concern:
```ruby
def self.sort(list)
  # Define the two iterators required by TSort
  reversed = list.reverse
  each_node = ->(&b) { reversed.each(&b) }
...
```